fix.py

```
@@ -0,0 +1,121 @@
+
+@app.post("/webhook")
+async def github_webhook(request: Request, x_hub_signature_256: str = Header(None)):
+    payload_body = await request.body()
+    if not verify_signature(payload_body, x_hub_signature_256):
+        raise HTTPException(status_code=401, detail="Invalid signature")
+
+    event = request.headers.get("X-GitHub-Event")
+    try:
+        data = json.loads(payload_body)
+    except json.JSONDecodeError:
+        raise HTTPException(status_code=400, detail="Invalid JSON")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)